Define some more terms/acronyms. (#610)
riscv/riscv-debug-spec: introduction.tex

debugging tasks. However, in many scenarios, hardware support is essential.

This document outlines a standard architecture for debug support
on RISC-V hardware platforms. This architecture allows a variety of implementations and
tradeoffs, which is complementary to the wide range of RISC-V implementations.
At the same time, this specification defines common interfaces to
allow debugging tools and components to target a variety of hardware
platforms based on the RISC-V ISA.

System designers may choose to add additional hardware debug support,
but this specification defines a standard interface for common
functionality.

\section{Terminology}

\begin{description}[style=nextline]
    \item[AMO]
        Atomic Memory Operation.
    \item[BYPASS]
        JTAG instruction that selects a single bit data register, also called BYPASS.
    \item[component]
        A RISC-V core, or other part of a hardware platform.

>         Typically all components will all be connected to a single system
>         bus.

Reviewer: Not a big deal but:
```suggestion
        Typically all components will be connected to a single system
        bus.
```

    \item[CSR]
        Control and Status Register.
    \item[DM]
        Debug Module (see Section \ref{dm}).
    \item[DMI]
        Debug Module Interface (see Section \ref{dmi}).
    \item[DR]
        JTAG Data Register.
    \item[DTM]
        Debug Transport Module (see Section \ref{dtm}).
    \item[DXLEN]
        Debug XLEN, which is the widest XLEN a hart supports, ignoring the
        current value of \Fmxl in \Rmisa.
    \item[GPR]
        General Purpose Register.
    \item[hart]
        A hardware thread in a RISC-V core.
    \item[IDCODE]
        32-bit Identification CODE, and a JTAG instruction that returns the IDCODE value.
    \item[IR]
        JTAG Instruction Register.
    \item[JTAG]
        Refers to work done by IEEE's Joint Test Action Group, described in
        IEEE 1149.1.
    \item[NAPOT]
        Naturally Aligned Powers-Of-Two.
    \item[NMI]
        Non-Maskable Interrupt.
    \item[hardware platform]

>         A single system consisting of one or more \emph{components}.

Reviewer 2: The rest of this list is alphabetized.  When this item was changed from `platform` to `hardware platform`, it is now in the wrong place.

\section{Terminology}

\begin{description}[style=nextline]
    \item[AMO]
        Atomic Memory Operation.
    \item[BYPASS]
        JTAG instruction that selects a single bit data register, also called BYPASS.
    \item[component]
        A RISC-V core, or other part of a platform.
        Typically all components will all be connected to a single system
        bus.
    \item[CSR]
        Control and Status Register.
    \item[DM]
        Debug Module (see Section \ref{dm}).
    \item[DMI]
        Debug Module Interface (see Section \ref{dmi}).
    \item[DR]
        JTAG Data Register.
    \item[DTM]
        Debug Transport Module (see Section \ref{dtm}).
    \item[DXLEN]
        Debug XLEN, which is the widest XLEN a hart supports, ignoring the
        current value of \Fmxl in \Rmisa.
    \item[GPR]
        General Purpose Register.
    \item[hart]
        A hardware thread in a RISC-V core.
    \item[IDCODE]
        32-bit Identification CODE, and a JTAG instruction that returns the IDCODE value.
    \item[IR]
        JTAG Instruction Register.
    \item[JTAG]

>         Joint Test Action Group, referring to IEEE 1149.

Reviewer 2: ```suggestion
        Joint Test Action Group, referring to IEEE 1149.1.
```

Author: I consciously didn't include .1, thinking that .7 should also be included. But maybe .7 really is cJTAG only?

Author: How about "Refers to work done by IEEE's Joint Test Action Group, described in IEEE 1149.1."

Reviewer 2: That sounds good.  The Debug Spec DTM section only describes an 1149.1 implementation, so it seems correct to refer to that as opposed to "1149".

    \item[NAPOT]
        Naturally Aligned Powers-Of-Two.
    \item[NMI]
        Non-Maskable Interrupt.
    \item[hardware platform]

>         A single integrated circuit consisting of one or more \emph{components}.

Reviewer: I didn't notice this the first time but it doesn't seem like it's necessarily a single integrated circuit.  Multi-chip modules, multi-socket systems, etc.  Fifteen years ago I worked on a multi-board system with seamless cache coherence between cores on the same chip, cores on different chips on the same board, and cores on chips on different boards.

Author: Sure. So what do I call it? A single system? A thing?

Reviewer: I'd go with system.

\section{Terminology}

\begin{description}[style=nextline]
    \item[AMO]
        Atomic Memory Operation.
    \item[BIST]

>         Built-In Self Test.

Reviewer: BIST is only used in one place in the document so it's probably easier to just say "BIST (built-in self test)" in that one location instead of having the reader refer to the glossary.

Author: Done.

    \item[BYPASS]
        JTAG instruction that selects a single bit data register, also called BYPASS.
    \item[component]

>         A RISC-V core, or other part of a platform.

Reviewer 3: ```suggestion
        A RISC-V core, or other part of a platform.
```

should this be core or hart?

Author: I agree with @pdonahue-ventana

        Typically all components will all be connected to a single system
        bus.
    \item[CSR]

>         Control Status Register.

Reviewer 3: ```suggestion
        Control/Status Register.
```

Isn't this an "and/or" type thing?

    \item[DM]

>         Debug Module.

Reviewer: Can this include a link to the DM chapter?  Same question for DMI and DTM and SBA.

Author: Done.

    \item[DMI]
        Debug Module Interface.
    \item[DR]

>         Data Register.

Reviewer 3: ```suggestion
        JTAG Data Register.
```
?

    \item[DTM]
        Debug Transport Module.
    \item[DXLEN]
        Debug XLEN, which is the widest XLEN a hart supports, ignoring the
        current value of \Fmxl in \Rmisa.
    \item[GPR]
        General Purpose Register.
    \item[hart]
        A hardware thread in a RISC-V core.
    \item[IDCODE]
        32-bit Identification CODE, and a JTAG instruction that returns the IDCODE value.
    \item[IR]

>         Instruction Register.

Reviewer 3: ```suggestion
        JTAG Instruction Register.
```

?

    \item[JTAG]
        Joint Test Action Group.
    \item[MIE]

>         Machine Interrupt Enable register; also a bit in mstatus.

Reviewer 3: ```suggestion
        Machine-Mode Interrupt Enable register; also a bit in mstatus.
```

When using `MIE` do you always mean the bit in `mstatus`? Would you use `mie` if you were talking about the actual register?

Reviewer: I agree on the distinction between MIE and mie.  (I find it confusing but I think it's too late now.)

Author: The privspec defines this register as "Machine interrupt-enable register." so -mode seems out of place. In this spec, we always style mie lowercase, and never refer to the bit in mstatus.

    \item[NAPOT]
        Naturally Aligned Powers-Of-Two.
    \item[NMI]
        Non-Maskable Interrupt.
    \item[OpenOCD]

>         Open On-Chip Debugger.

Reviewer: This is only used once in the document and given that context I don't this glossary entry is necessary.

Author: Done.

    \item[PC]
        Program Counter.
    \item[platform]

>         A single integrated circuit consisting of one or more \emph{components}.

Reviewer: Is this a "platform" or a "hardware platform"?

Author: Looking it up, this is a "hardware platform." But do we use that term anywhere?

Author: Based on this, I think I need to look at each use of "platform" and probably change it to "hardware platform." Do you agree?

Reviewer: I have always found it annoying and confusing that RISC-V uses the word "platform" both for a particular implementation and for standards such as the Unix platform.  But we're stuck with that so "hardware platform" seems appropriate.

Now that I'm looking, I notice that this spec uses both "platform" and "system" to describe the overall implementation:
- "ndmreset resets all the harts in the system"
- "Harts are nonexistent if they will never be part of this system"
- "In most platforms there will only be one DM"
- "There may be multiple DTMs in a single platform."

Is there a distinction?  Maybe "system" (like in the title of chapter 2) means the group of components involved in debug but not all of the components in the whole chip/board/whatever?  Or maybe there's really no distinction?  The base architecture also uses the word "system" so maybe I'm overthinking this.

Author: I don't think there's any intentional use of platform vs. system. They should probably all be "hardware platform."

    \item[physical address]
        An address that is directly usable on the system bus.
    \item[RV32]
        RISC-V core with XLEN of 32.
    \item[RV64]
        RISC-V core with XLEN of 64.
    \item[SBA]
        System Bus Access.
    \item[TAP]

>         Test Access Port.

Reviewer: This doesn't seem self-explanatory.  Maybe "see IEEE 1149" (or whatever) on both this and the JTAG definition.

    \item[XLEN]

>         Width of the core's integer register file.

Reviewer 3: ```suggestion
        Width of a hart's integer register file.
```

    \item[virtual address]

>         An address as a hart sees it. If there is address translation this

Reviewer 3: ```suggestion
        An address as a hart sees it. If the hart is using address translation this
```
